docs/http-authentication.rst

```diff
@@ -17,89 +18,23 @@ In this example, the username is test and the password is basic.
 ..  code-block:: pycon
 
     >>> import feedparser
-    >>> d = feedparser.parse('http://test:basic@feedparser.org/docs/examples/basic_auth.xml')
+    >>> d = feedparser.parse('http://test:basic@$READTHEDOCS_CANONICAL_URL/examples/basic_auth.xml')
```

Comment:
The previous URL was no longer available, so the example didn't work

I believe this should get expanded to https://feedparser.readthedocs.io/en/latest/examples/basic_auth.xml instead, which should work - though this feed doesn't actually seem to be protected by basic auth (and same for https://feedparser.readthedocs.io/en/latest/examples/digest_auth.xml), but I checked this works with a different URL that does have basic auth